### PR TITLE
Test for KafkaConnector recreation issue after invalid config update

### PR DESCRIPTION
**Pull request by michalxo**
### Type of change

Test for PR#6443 and Issue #6438

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Reference relevant issue(s) and close them after merging

### CODE REVIEW

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectIsolatedST.java [hunk 1]
```diff
@@ -1318,6 +1322,71 @@ void testKafkaConnectWithScramShaAuthenticationRolledAfterPasswordChanged(Extens
...
+            .editSpec()
+                .withTasksMax(2)
+                .withClassName("org.apache.kafka.connect.file.FileStreamSinkConnector")
+                .addToConfig("topics", storage.getTopicName())
+                .addToConfig("file", Constants.DEFAULT_SINK_FILE_PATH)
+            .endSpec()
+            .build());
+
+        String connectorName = storage.getClusterName();
+        String coExpectedLog = "KafkaConnect(" + storage.getNamespaceName() + "/" + connectorName + "): Configuration option \"tasks.max\" is forbidden and will be ignored";
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectIsolatedST.java:1358]: Its confusing if you call it _expected_ when it is expected in one case and unexpected in the other.

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectIsolatedST.java [hunk 2]
```diff
@@ -1318,6 +1322,71 @@ void testKafkaConnectWithScramShaAuthenticationRolledAfterPasswordChanged(Extens
...
+        String coNotExpectedLog = "KafkaConnect(" + storage.getNamespaceName() + "/" + connectorName + "): Configuration parameter tasks.max in KafkaConnector.spec.config will be ignored and the value from KafkaConnector.spec will be used instead";
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectIsolatedST.java:1359]: Where would this log message show? It does not seem to be in the code base.

**Review by scholzj — COMMENTED** [review]
I left some comments. The test looks good overall, but TBH I'm not sure it is worth having an ST for this.

File: systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectIsolatedST.java (hunk 2 above)
Comment (michalxo) [reply systemtest/src/test/java/io/strimzi/systemtest/connect/ConnectIsolatedST.java:1359]: It used to be in previous (older builds of CO). Should I remove that line then?

**Comment by michalxo** [comment]
Closing this PR as not needed after discussion offline.

**Pull request closed without merging**